File: java/src/test/java/com/haibo/yan/algorithm/stack/TestMinStack.java
package com.haibo.yan.algorithm.stack;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

@Test
public class TestMinStack {


    public void testMin() {
        MinStack<Integer> minStack = new MinStack<>();

        minStack.push(3);
        minStack.push(1);
        minStack.push(5);

        assertEquals(minStack.getMin(), Integer.valueOf(1));
        assertEquals(minStack.top(), Integer.valueOf(5));

        minStack.push(0);

        assertEquals(minStack.getMin(), Integer.valueOf(0));
        assertEquals(minStack.top(), Integer.valueOf(0));

        minStack.pop();

        assertEquals(minStack.getMin(), Integer.valueOf(1));
        assertEquals(minStack.top(), Integer.valueOf(5));
    }
}

File: python/test/datastructure/test_min_stack.py
from unittest import TestCase
from datastructure.min_stack import MinStack

class TestMinStack(TestCase):
    def test_stack(self):
        stack = MinStack()

        stack.push(3)
        stack.push(1)
        stack.push(5)

        self.assertEqual(1, stack.min())
        self.assertEqual(5, stack.top())

        stack.push(0)

        self.assertEqual(0, stack.min())
        self.assertEqual(0, stack.top())

        stack.pop()
        self.assertEqual(1, stack.min())
        self.assertEqual(5, stack.top())
